src/Serializer/ItemNormalizer.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Serializer;
+
+/**
+ * Generic item normalizer.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ItemNormalizer extends AbstractItemNormalizer
+{
+    public function normalize($object, $format = null, array $context = [])
+    {
+        $rawData = parent::normalize($object, $format, $context);
+        if (!is_array($rawData)) {
+            return $rawData;
+        }
+
+        if (!isset($data['id'])) {
```

Comment:
IMO it's better to always use an IRI as identifier but I'm open to discussion :)